x/twap/README.md

```diff
@@ -12,6 +12,22 @@ Notice that the latest price `p_n` isn't used, as it has lasted for a time inter
 To illustrate with an example, given the sequence: `(0s, $1), (4s, $6), (5s, $1)`, the arithmetic mean TWAP is: 
 $$\frac{\$1 * (4s - 0s) + \$6 * (5s - 4s)}{5s - 0s} = \frac{\$10}{5} = \$2$$
 
+## Geometric mean TWAP
+
+While the arithmetic mean TWAPs are much more widely used, they should theoretically be less accurate in measuring a geometric Brownian motion process (which is how price movements are usually modeled)
+
+Arithmetic TWAP tends to overweight higher prices relative to lower ones.
+
+Therefore, we also support a geometric mean TWAP.
+
+The core functionality stays similar to the arithmetic mean TWAP. However, instead of computing the geometric mean TWAP naively, we use the following property:
+
+$$GeometricMean(P) = 1.0001^{ArithmeticMean(log_{1.0001}{P})}$$
+
+Naive computation is expensive and easily overflows. As a result, we track logarithms of prices instead of prices themselves in the accumulators.
+When geometric twap is requested, we first compute the arithmetic mean of the logarithms, and then exponentiate it with the same base as the logarithm
+to get the final result.
```

Comment:
Can we add a line or two about why this holds? 

IIUC, it's basically:

$$  {(\prod_{i=a}^{b} P_i)}^{\frac{1}\{b-a}}  =  exp(\sum_{i=a}^{b}{\frac{1}{b-a} log_{1.0001}{(P_i)}}) = exp(log_{1.0001}{(\frac{1}{b-a}\sum{P_i})})$$

Reply:
Just realized this is the case by definition. So maybe we don't need it

Reply:
The only difference is that, currently, we take 1.0001 to that power, not Euler's number. It might change to a larger value down the road due to the risk of overflows with such a small base for the logarithm. A small base leads us to have a large exponent in `Power()` function that might overflow

Yeah while I tried to explain in words, I agree that it's nice to have the formula.

I didn't add it in the first place because having so many issues with rendering math on GitHub 90% of the time they get rendered on Notion but not here 😢

Reply:
The base of `1.0001` wouldn't format so I added this:

https://github.com/osmosis-labs/osmosis/pull/3420/commits/6fd661d33bd9ce8ae44a711fcb95410acef4c821

Reply:
yeah, I took `exp()` to represent $$exp_{1.0001}()$$ here just to avoid writing it. That makes sense (and the comment on the new commit looks great!)